lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1994,6 +1994,12 @@ sub done {
         my $auto_clone_regex = OpenQA::App->singleton->config->{global}->{auto_clone_regex};
         $restart = 1 if $result eq INCOMPLETE && $auto_clone_regex && $reason =~ $auto_clone_regex;
 
+        # for poo#80334
+        my $autoinst_log_file = path($self->result_dir, "autoinst-log.txt");
+        my $command           = "grep \"No space left on device\" $autoinst_log_file > /dev/null";
+        $restart = 1
+          if ($restart == 0 && $reason =~ /terminated prematurely with corrupted state file/ && system($command) == 0);
```

Comment:
It would be better to grep within the log file directly from the Perl script instead of starting a shell and grep. I would also avoid any extra checks unless the result is incomplete.

Reply:
```perl
open my $fh, "<", $autoinst_log_file or die "Failed to open $autoinst_log_file";
my @lines = sort grep /No space left on device/, <$fh>
```

Something like this 🤔

Reply:
> It would be better to grep within the log file directly from the Perl script instead of starting a shell and grep. 

Could you give me more details?  read content from the file and do grep? just like @kalikiana wrote. Thanks.